### PR TITLE
NET-147 full config for nm-quick.sh

### PR DESCRIPTION
**Pull request by TobiaszCudnik**
## Describe your changes

NET-147 Config file support for nm-quick.sh

Changes:
- single config file `netmaker.env`
  - added reference comments
- env vars fed to compose & caddyfile
  - via an `.env` symlink
- substitutions handled by docker-compose
  - eg `BROKER_ENDPOINT=wss://broker.${NM_DOMAIN}`
- reduced docker-compose files
  - `compose/docker-compose.yml` main dist, added reference comments
  - `compose/docker-compose.ee.yml` extends the main one (via an override)
  - `compose/docker-compose-emqx.yml` extends the main one (via an override)
- relative install path - to the location of `nm-quick.sh` (not `pwd`)
  - different install dir == separate containers/volumes (no migrations at this point)
- error handling, cleanup calls
- fixed nm-certs for non-local installs
- added debug vars to speed things up
  - `NM_SKIP_BUILD`, `NM_SKIP_CLONE`, `NM_SKIP_DEPS`

## Provide testing steps

```
wget https://raw.githubusercontent.com/gravitl/netmaker/NET-147/config-compose-caddy/scripts/nm-quick.sh
chmod +x nm-quick.sh
./nm-quick.sh -b local -t NET-147/config-compose-caddy
```

### Known issues
- EE version breaks on MQ URL

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [x] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

**Comment by TobiaszCudnik** [comment]
> Since the subdomain of each server is a fixed string of api, broker, turn, etc, I'd like to be able to customize it.

@miyu4u Thats a valid request, unfortunately its out of scope for this PR which tries to avoid new features. I will take a note of it and try to handle it in the next iteration, thanks.

File: compose/docker-compose.yml [hunk 1]
```diff
@@ -1,71 +1,69 @@
 version: "3.4"
 
 services:
+
   netmaker:
     container_name: netmaker
-    image: gravitl/netmaker:REPLACE_SERVER_IMAGE_TAG
+    image: gravitl/netmaker:$SERVER_IMAGE_TAG
```
Comment (afeiszli) [inline compose/docker-compose.yml:7]: should $SERVER_IMAGE_TAG be wrapped in brackets like the other vars? e.g. ${SERVER_IMAGE_TAG}

File: compose/docker-compose.yml [hunk 2]
```diff
@@ -1,71 +1,69 @@
...
+      - TURN_SERVER_HOST=turn.${NM_DOMAIN}
+      # domain of the turn api server
+      - TURN_SERVER_API_HOST=https://turnapi.${NM_DOMAIN}
     ports:
       - "3478:3478/udp"
+
   netmaker-ui:
     container_name: netmaker-ui
-    image: gravitl/netmaker-ui:REPLACE_UI_IMAGE_TAG
+    image: gravitl/netmaker-ui:$UI_IMAGE_TAG
```
Comment (afeiszli) [inline compose/docker-compose.yml:34]: same question, brackets?

File: compose/docker-compose.yml [hunk 3]
```diff
@@ -74,36 +72,36 @@ services:
...
-      PASSWORD: "REPLACE_TURN_PASSWORD"
+
 volumes:
-  caddy_data: {}
-  caddy_conf: {}
-  sqldata: {}
-  dnsconfig: {}
-  mosquitto_logs: {}
-  turn_server: {}
+  caddy_data: { } # runtime data for caddy
```
Comment (afeiszli) [inline compose/docker-compose.yml:101]: Why the space in brackets here?

**Comment by theguy951357** [comment]
will test when changes addressed.

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes

File: compose/docker-compose.yml (hunk 3 above)
Comment (TobiaszCudnik) [reply compose/docker-compose.yml:101]: Auto formater did.

File: compose/docker-compose.yml (hunk 1 above)
Comment (TobiaszCudnik) [reply compose/docker-compose.yml:7]: Both are supported ([docs](https://docs.docker.com/compose/compose-file/compose-file-v3/#variable-substitution)). Ive added a TODO for error handling to NET-155, these minor differences can be handled there as well.

**Review by afeiszli — APPROVED** [review]

**Pull request merged**